src/ansi.rs

```diff
@@ -1,18 +1,163 @@
-use std::borrow::Cow;
+use std::{
+    borrow::Cow,
+    iter::{FusedIterator, Peekable},
+    str::CharIndices,
+};
 
-use once_cell::sync::Lazy;
-use regex::{Matches, Regex};
+#[derive(Debug, Clone, Copy)]
+enum State {
+    Start,
+    S1,
+    S2,
+    S3,
+    S4,
+    S5,
+    S6,
+    S7,
+    S8,
+    S9,
+    S10,
+    S11,
+    Trap,
+}
 
-static STRIP_ANSI_RE: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(
-        r"[\x1b\x9b]([()][012AB]|[\[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-PRZcf-nqry=><])",
-    )
-    .unwrap()
-});
+impl Default for State {
+    fn default() -> Self {
+        Self::Start
+    }
+}
+
+impl State {
+    fn is_final(&self) -> bool {
+        use State::*;
```

Comment:
I would prefer we use `State::S1` and friends here instead of using these. I was bitten more than once with a pattern accidentally turning into a variable match.